packages/connectivity/android/src/main/java/io/flutter/plugins/connectivity/ConnectivityMethodChannelHandler.java

```diff
@@ -0,0 +1,163 @@
+package io.flutter.plugins.connectivity;
+
+import android.content.BroadcastReceiver;
+import android.content.Context;
+import android.content.Intent;
+import android.content.IntentFilter;
+import android.net.ConnectivityManager;
+import android.net.Network;
+import android.net.NetworkCapabilities;
+import android.net.NetworkInfo;
+import android.net.wifi.WifiInfo;
+import android.net.wifi.WifiManager;
+import android.os.Build;
+import io.flutter.plugin.common.EventChannel;
+import io.flutter.plugin.common.MethodCall;
+import io.flutter.plugin.common.MethodChannel;
+
+public class ConnectivityMethodChannelHandler
+    implements MethodChannel.MethodCallHandler, EventChannel.StreamHandler {
+
+  private final Context context;
+  private ConnectivityManager manager;
+  private BroadcastReceiver receiver;
+
+  public ConnectivityMethodChannelHandler(Context context) {
+    assert (context != null);
+    this.context = context;
+    this.manager =
+        (ConnectivityManager)
+            context.getApplicationContext().getSystemService(Context.CONNECTIVITY_SERVICE);
+  }
+
+  @Override
+  public void onListen(Object arguments, EventChannel.EventSink events) {
```

Comment:
Please add nullability annotations to all input parameters and return values for each public and private method.